datafusion/core/src/datasource/physical_plan/parquet/access_plan.rs

```diff
@@ -182,6 +183,11 @@ impl ParquetAccessPlan {
     /// is returned for *all* the rows in the row groups that are not skipped.
     /// Thus it includes a `Select` selection for any [`RowGroupAccess::Scan`].
     ///
+    /// # Errors
+    ///
+    /// Returns an error if the specified row selection does not specify
+    /// the same number of rows as in `row_group_metadata`.
```

Comment:
```suggestion
    /// Returns an error if any specified row selection does not specify
    /// the same number of rows as in it's corresponding `row_group_metadata`.
```

Is it also worth making a note that it's up to the caller to validate that the size of the ParquetAccessPlan matches the number of row groups?